### ISSUE TITLE
Terraform v0.12.3 - Syntax check broken: Unknown token IDENT

### ISSUE THREAD

**Issue by BB-7** [hashicorp/vscode-terraform#195]
Throughout our Terraform catalog of modules we are seeing syntax errors for "unknown token IDENT".

Examples:

![ident1](https://user-images.githubusercontent.com/43642511/60597431-eb403200-9d78-11e9-8258-58776582d7db.JPG)

![ident2](https://user-images.githubusercontent.com/43642511/60597534-16c31c80-9d79-11e9-8e44-41939a9a4027.JPG)

![ident3](https://user-images.githubusercontent.com/43642511/60597619-39553580-9d79-11e9-939e-2d07e77add0b.JPG)

It's not just local to my machine. Our entire team is experiencing it. We are on TF version v0.12.3.

**Comment by Danktified** [comment]
This is because of unfinished support for TF 0.12
#157

**Comment by sujaypillai** [comment]
Same here for me for a valid syntax according to v0.12 - 
<img width="697" alt="Screen Shot 2019-07-04 at 10 19 03 AM" src="https://user-images.githubusercontent.com/3936714/60635696-3411ec80-9e46-11e9-835c-3723f1a85932.png">

**Comment by davisford** [comment]
The only IDE I know that supports TF 0.12+ HCL syntax is [Intellij IDEA with the Terraform plugin](https://plugins.jetbrains.com/plugin/7808-hashicorp-terraform--hcl-language-support)

**Comment by fluffy-cakes** [comment]
Any update on syntax for v.12?

**Comment by HighwayofLife** [comment]
Enable the experimental Terraform language server, which has support for Terraform 0.12...
`settings.json`
```
    "terraform.languageServer": {
        "enabled": true,
        "args": []
    }
```

**Comment by fluffy-cakes** [comment]
> Enable the experimental Terraform language server, which has support for Terraform 0.12...
> `settings.json`
> 
> ```
>     "terraform.languageServer": {
>         "enabled": true,
>         "args": []
>     }
> ```

@HighwayofLife  This looks like it conflicts with this add-in. How do you have yours configured?

**Comment by karthiksadhasivam** [comment]
> Enable the experimental Terraform language server, which has support for Terraform 0.12...
> `settings.json`
> 
> ```
>     "terraform.languageServer": {
>         "enabled": true,
>         "args": []
>     }
> ```

Seems to have worked.

**Comment by HighwayofLife** [comment]
@fluffy-cakes 
> @HighwayofLife This looks like it conflicts with this add-in. How do you have yours configured?

That is how I have mine configured. You need vscode-terraform 1.4, which added support for Terraform 0.12 via a language server.

**Comment by rnsc** [comment]
> > Enable the experimental Terraform language server, which has support for Terraform 0.12...
> > `settings.json`
> > ```
> >     "terraform.languageServer": {
> >         "enabled": true,
> >         "args": []
> >     }
> > ```
> 
> Seems to have worked.

It brought another set of errors when using the "count" value in my TF files unfortunately :(

**Comment by skupjoe** [comment]
Language server didn't fix all of these for me. And brought some others. Is there any way to toggle language checking entirely?

**Comment by eponymz** [comment]
> Language server didn't fix all of these for me. And brought some others. Is there any way to toggle language checking entirely?

@skupjoe add this bad boy to your settings.json file
```json
  "terraform.indexing": {
    "enabled": false,
    "liveIndexing": false
  }
```

**Comment by skupjoe** [comment]
Worked with a reload thanks!

Shame to leave the indexing support, but I can wait for now. Thanks for the great plugin!

**Comment by cima** [comment]
Yep _terraform.languageServer_ and _terraform.indexing_ to false blocks did "green out" my code with terraform 0.12.x syntax features.

**Comment by MRostanski** [comment]
@cima How did you get rid of `count.index` errors?

**Comment by evandropomatti** [comment]
The problem is still happening. How much time will we have to live with the workaround?

**Comment by robguinness** [comment]
The workaround described actually created more issues than I had originally. Any updates on this would be most appreciated!

**Comment by Mentioum** [comment]
For those who are struggline to make this work a summary of the above:

Adding the following to your settings.json file:
```json
"terraform.languageServer": {
        "enabled": true,
        "args": []
    }
```
and 
```json
"terraform.indexing": {
    "enabled": false,
    "liveIndexing": false
  }
```
This tells VSCode to turn off the old extension's system and on the new language server system.

Restart VSCode once and it will prompt to download the language server.

Restart again after the install completes (if necessary)

Not saying this is OK or a good dev experience, but the solution in this thread does work. Looking forward to the language server becoming a mainstay :+1:

**Comment by gowrishcode** [comment]
Thank you Mentioum, Issue has been resolved. :-)

**Comment by lpzdvd-packlink** [comment]
Not sure if it's really a good idea. Currently working on gcp provider 3.x versions and I've seen language server latest version install gcp 2.x.

Warnings removed anyway.

**Comment by oceaneLonneux** [comment]
I think we all appreciate the effort for basically un-breaking the IDE, but it is true that a lot of "nice" features have been lost. For example, references don't work anymore, there is no link to the documentation anymore, ... 

This is not to complain about the work already done (linked to https://github.com/mauve/vscode-terraform/issues/157), but just maybe a reminder...? To be honest, I would gladly help but I'm afraid I don't even know where to look

**Comment by Pijuli** [comment]
Last contribution 1 Oct 2019. Is it dead?  :(

**Comment by dminca** [comment]
yea, so I guess we should look for another plugin since this one didn't receive any updates on Terraform syntax since last issue was reported on June 2019...

**Comment by BenjaminMtDew** [comment]
> > Enable the experimental Terraform language server, which has support for Terraform 0.12...
> > `settings.json`
> > ```
> >     "terraform.languageServer": {
> >         "enabled": true,
> >         "args": []
> >     }
> > ```
> 
> @HighwayofLife This looks like it conflicts with this add-in. How do you have yours configured?

Thank you. VS Code suggested I do this, but the information on how was not readily available.

**Comment by paultyng** [comment]
We just released [v2.0.0-rc.1](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) of the extension. The main features include:

* Added syntax support for 0.12
* Added [terraform-ls](https://github.com/hashicorp/terraform-ls) usage by default (currently on 0.3.0, which offers basic provider code completion)

You can find additional information and specifics in the [release notes](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) and [CHANGELOG](https://github.com/hashicorp/vscode-terraform/blob/master/CHANGELOG.md).

With this release we expect that many of the prior issues and PRs are no longer relevant or have been addressed, and are therefore being closed. If you feel the action taken on an issue or PR is in error, please comment as such and we can figure out the appropriate way to address it.

We plan to add the final 2.0.0 release to the marketplace soon, but are actively seeking your feedback now on the release candidates. You can download the .vsix from the releases page and [manually install it in VS Code](https://code.visualstudio.com/docs/editor/extension-gallery#_install-from-a-vsix) to try it out.

**Comment by triwats** [comment]
What a fanastic time on my part to check if there's any progress on this. I just installed the extension manually as mentioned above and the annoyance with faulty syntax highlighting has indeed gone. Thanks for all the hard work @paultyng, @aeschright et al. Looking forward to this becoming mainline in the future.
